src/test/java/net/mcreator/integration/workspace/ReferencesFinderTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.integration.workspace;
+
+import net.mcreator.generator.Generator;
+import net.mcreator.generator.GeneratorConfiguration;
+import net.mcreator.generator.GeneratorFlavor;
+import net.mcreator.integration.TestSetup;
+import net.mcreator.integration.TestWorkspaceDataProvider;
+import net.mcreator.integration.generator.GTModElements;
+import net.mcreator.integration.generator.GTSampleElements;
+import net.mcreator.integration.ui.UITestUtil;
+import net.mcreator.minecraft.ElementUtil;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.dialogs.SearchUsagesDialog;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.util.ListUtils;
+import net.mcreator.workspace.ReferencesFinder;
+import net.mcreator.workspace.Workspace;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.elements.SoundElement;
+import net.mcreator.workspace.elements.VariableElement;
+import net.mcreator.workspace.resources.Model;
+import net.mcreator.workspace.settings.WorkspaceSettings;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ReferencesFinderTest {
+
+	private static Logger LOG;
+	private static Random random;
+
+	private static Workspace workspace;
+	private static MCreator mcreator;
+
+	@BeforeAll public static void initTest() throws IOException {
+		System.setProperty("log_directory", System.getProperty("java.io.tmpdir"));
+		LOG = LogManager.getLogger("References Finder Test");

```

Comment:
If it is a static variable, you can define it directly when initializing it as we do everywhere else in the code.

Reply:
We do that in usual code, but not in tests where we first need to set the logging directory (check how other tests do that).